File: frontend/src/api/signal.ts
import client from './client'

export interface SignalItem {
  id: number
  code: string
  stock_name: string
  trade_date: string
  direction: string
  score: number
  reason: string
  close_price: number
  suggested_buy_low: number
  suggested_buy_high: number
  stop_loss_price: number
  target_price: number
}

export function fetchTodaySignals() {
  return client.get<SignalItem[]>('/signal/today')
}

export function fetchHistory(days = 30, direction?: string) {
  const params: Record<string, unknown> = { days }
  if (direction) {
    params.direction = direction
  }
  return client.get<SignalItem[]>('/signal/history', { params })
}

File: backend/app/schemas/signal.py
from datetime import date
from pydantic import BaseModel


class SignalItem(BaseModel):
    id: int
    code: str
    stock_name: str
    trade_date: date
    direction: str
    score: int
    reason: str
    close_price: float
    suggested_buy_low: float | None
    suggested_buy_high: float | None
    stop_loss_price: float | None
    target_price: float | None

    class Config:
        from_attributes = True
